src/components/Layouts/Primary/NavBar/NavBar.tsx

```diff
@@ -0,0 +1,208 @@
+
+interface NavBarProps {
+  accountListId: string;
+  onMobileClose: () => void;
+  openMobile: boolean;
+}
+
+interface Item {
+  href?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon?: any;
+  items?: Item[];
+  title: string;
+}
+
+interface Section {
+  as?: string;
+  href?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  icon?: any;
+  items?: Item[];
+  title: string;
+}
+
+const sections: Section[] = [
+  {
+    title: 'Dashboard',
+    href: '/accountLists/[accountListId]',
+  },
+  {
+    title: 'Contacts',
+    href: '/accountLists/[accountListId]/contacts',
+  },
+  {
+    title: 'Tasks',
+    href: '/accountLists/[accountListId]/tasks',
+  },
+  {
+    title: 'Reports',
+    items: ReportNavItems.map((item) => ({
+      ...item,
+      title: item.subTitle ? `${item.title} (${item.subTitle})` : item.title,
+      href: `/accountLists/[accountListId]/reports/${item.id}`,
+    })),
+  },
+  {
+    title: 'Tools',
+    items: ToolsList.flatMap((toolsGroup) => [
+      ...toolsGroup.items.map((tool) => ({
+        title: tool.tool,
+        href: `/accountLists/[accountListId]/reports/${tool.id}`,
+      })),
+    ]),
+  },
+  {
+    title: 'Coaches',
+    href: '/coaches',
+  },
+];
+
+function renderNavItems({
+  accountListId,
+  items,
+  pathname,
+  depth = 0,
+}: {
+  accountListId: string;
+  items: Item[];
+  pathname: string;
+  depth?: number;
+}) {
+  return (
+    <List disablePadding>
+      {items?.reduce<ReactElement[]>(
+        (acc, item) =>
+          reduceChildRoutes({
+            acc,
+            accountListId,
+            item,
+            pathname,
+            depth,
+          }),
+        [],
+      )}
+    </List>
+  );
+}
+
+function reduceChildRoutes({
+  acc,
+  accountListId,
+  pathname,
+  item,
+  depth,
+}: {
+  acc: ReactElement[];
+  accountListId: string;
+  pathname: string;
+  item: Item;
+  depth: number;
+}) {
+  const key = item.title + depth;
+  const { t } = useTranslation();
+
+  if (item.items) {
+    acc.push(
+      <NavItem depth={depth} icon={item.icon} key={key} title={t(item.title)}>
+        {renderNavItems({
+          accountListId,
+          depth: depth + 1,
+          pathname,
+          items: item.items,
+        })}
+      </NavItem>,
+    );
+  } else {
+    acc.push(
+      <NavItem
+        depth={depth}
+        href={item.href}
+        as={
+          accountListId && item.href?.includes('[accountListId]')
+            ? item.href.replace('[accountListId]', accountListId)
+            : item.href
+        }
+        icon={item.icon}
+        key={key}
+        title={t(item.title)}
+      />,
+    );
+  }
+
+  return acc;
+}
+
+const useStyles = makeStyles((theme: Theme) => ({
+  mobileDrawer: {
+    width: 290,
+    backgroundColor: theme.palette.cruGrayDark.main,
+    zIndex: theme.zIndex.drawer + 200,
+  },
+}));
+
+export const NavBar: FC<NavBarProps> = ({
+  accountListId,
```

Comment:
This can be a hook in the `NavBar` component, it just has to be called in the component function itself and not after any conditions that would return early or in helper functions.